rootfs/api/models/app.py

```diff
@@ -457,14 +496,23 @@ def deploy(self, release):
                 'build_type': release.build.type,
                 'healthcheck': release.config.healthcheck,
                 'routable': routable,
-                'deploy_batches': batches
+                'deploy_batches': batches,
+                'deploy_timeout': 120,  # 2 minutes
+                'deployment_history_limit': deployment_history,
+                'deployments': deployments,
+                'release_summary': release.summary
             }
 
         # Sort deploys so routable comes first
         deploys = OrderedDict(sorted(deploys.items(), key=lambda d: d[1].get('routable')))
 
         for scale_type, kwargs in deploys.items():
             try:
+                # Is there an existing deployment in progress?
+                name = self._get_job_id(scale_type)
+                if not force_deploy and release.deployment_in_progress(self.id, name):
```

Comment:
Nope, haven't thought about it for the overall Controller. But it's fine as an in-memory boolean since it is checking the state of the Deployment object (owned by Kubernetes) so there is no stateful information to be concerned about for 2+ replicas of the controller as they all hit the same source of truth (kubernetes)